### PR TITLE
feat(Image): modify layout and support closeIcon、PreviewGroup fallback

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/react-component/image/issues/263
https://github.com/react-component/image/issues/157

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     modify layout style, support preview closeIcon and PreviewGroup fallbak，fixed an issue where mount would load preview resource.      |
| 🇨🇳 Chinese |     修改布局风格，preview 支持 closeIcon，PreviewGroup 支持 fallback，修复一开始会加载预览资源的问题。     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at b8148cf</samp>

Added new features to the `Image.PreviewGroup` component to enable users to flip and rotate the image and download the modified version. Updated the `zh-CN` documentation for the component in `components/image/demo/toolbarRender.md`.

### <samp>🤖 Generated by Copilot at b8148cf</samp>

* Add new buttons for downloading the original and transformed images in the toolbar of the `Image.PreviewGroup` component ([link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-ff24c471c0b5319a02b8f90d81007f652989af18a8555822ca98d324c8b00d73L3-R7), [link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852L2-R2), [link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852R68), [link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852L42-R96))
* Rename the existing download function to `onOriginalImgDownload` and change the file extension to `png` ([link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852L8-R8), [link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852L15-R15))
* Implement a new function `onTransformImgDownload` that creates a canvas, flips and rotates the image, and downloads it as a `png` file ([link](https://github.com/ant-design/ant-design/pull/43167/files?diff=unified&w=0#diff-94187e7b726a3e17e6c68962179363e372b1b38b03c72eb8d78cc5cf2bff3852R23-R60))

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/image/demo/toolbarRender.tsx [hunk 1]
```diff
@@ -39,9 +78,22 @@ const App: React.FC = () => {
...
           <ul className="ant-image-preview-operations">
-            <li className="ant-image-preview-operations-operation" onClick={onDownload}>
-              <DownloadOutlined className="ant-image-preview-operations-icon" />
-            </li>
+            <Tooltip title="download the original image" zIndex={9999}>
+              <li
+                className="ant-image-preview-operations-operation"
+                onClick={onOriginalImgDownload}
+              >
+                <DownloadOutlined className="ant-image-preview-operations-icon" />
```
Comment (afc163) [inline components/image/demo/toolbarRender.tsx:86]: ant-image-preview-operations-operation 和 ant-image-preview-operations-icon 这两得这么写死比较奇怪，能拿到就好了。

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx:86]: 没办法，toolbarRender 太自由了，如果只想添加一个图标，只能用原来的 class 来保持原来的样式，这个之前讨论过了呀，目测没有更好的办法。

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx:86]: 从参数里暴露出来 classname 会不会太奇怪了，没有先前的 api 有这么做过

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx:86]: className: { ulClassName, liClassName, iconClassName } ? 😂

Comment (afc163) [reply components/image/demo/toolbarRender.tsx:86]:
```
classes: { operations, operation, operationIcon }
```

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx:86]: 那就变成这样了，让其他人看看 👀
image

Comment (zombieJ) [reply components/image/demo/toolbarRender.tsx:86]: 可以参考一下 https://github.com/ant-design/ant-design/discussions/42833，提供 `components` 抽象：

```tsx
interface components {
  Operation: ReactComponent;
}
```

用户消费的时候就拿这个包 Icon：

```tsx
<Operation onClick={...}>
  <SmileOutlined />
</Operation>
```

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx:86]: 这样 SmileOutlined 还是要加 ant-image-preview-operations-icon 吧？

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx]: 保持现状，demo 改成这样。用户如果还是要用 ant 的 class 就是他们自己的事了
image

Comment (afc163) [reply components/image/demo/toolbarRender.tsx]: 好看~

Comment (afc163) [reply components/image/demo/toolbarRender.tsx]: 可以用这个当默认样式么？

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx]: 不好吧，这个设计变了感觉算是 breaking change 了吧，现在完全可以让用户根据他们的 ui 设计来自定义了呀

Comment (afc163) [reply components/image/demo/toolbarRender.tsx]: Ant Design 是设计规范，也要向前迭代的。

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx]: 之前挺多人用 countRender 或者其它办法添加下载按钮到上方的，改了设计就坏掉了😂

Comment (zombieJ) [reply components/image/demo/toolbarRender.tsx]: 好看，我带给设计师去

Comment (afc163) [reply components/image/demo/toolbarRender.tsx]: 我们可以内建一个新的样式类型。

**Comment by zombieJ** [comment]
来了：
image

**Comment by linxianxi** [comment]
image
image
image

File: package.json [hunk 2]
```diff
@@ -129,7 +129,7 @@
     "rc-drawer": "~6.2.0",
     "rc-dropdown": "~4.1.0",
     "rc-field-form": "~1.33.0",
-    "rc-image": "~6.1.0",
+    "rc-image": "~7.0.0-2",
```
Comment (kiner-tang) [inline package.json:132]: 如果验证没问题的话，rc-image 是不是要发一个正式的 7.0.0 版本？

Comment (linxianxi) [reply package.json:132]: @zombieJ

**Comment by linxianxi** [comment]
看看样式有哪些需要调整的

File: package.json (hunk 2 above)
Comment (zombieJ) [reply package.json:132]: 没事儿，这个合进去我发正式版，也是会提升上来的。

**Comment by afc163** [comment]
图标 hover 没有 transition 。

**Comment by linxianxi** [comment]
> 图标 hover 没有 transition 。

加了

File: components/image/demo/toolbarRender.tsx [hunk 3]
```diff
@@ -1,58 +1,98 @@
...
-            {zoomOutIcon}
-            {zoomInIcon}
-            {closeIcon}
-          </ul>
+          <Space size={12} className="toolbar-wrapper">
+            <Tooltip title="download the original image" zIndex={9999}>
+              <DownloadOutlined onClick={onOriginalImgDownload} />
+            </Tooltip>
+            <Tooltip title="download the flipped and rotated image" zIndex={9999}>
+              <DownloadOutlined onClick={() => onTransformImgDownload({ flipY, flipX, rotate })} />
```
Comment (zombieJ) [inline components/image/demo/toolbarRender.tsx:85]: 这个例子可以简化一下，就是提供一个下载按钮就好了。东西多了反而不知道要抄啥。

File: components/image/demo/toolbarRender.tsx [hunk 4]
```diff
@@ -1,58 +1,95 @@
...
+  //     ctx.drawImage(img, -naturalWidth / 2, -naturalHeight / 2);
+
+  //     const link = document.createElement('a');
+  //     link.href = canvas.toDataURL('image/png');
+  //     link.download = 'image.png';
+  //     document.body.appendChild(link);
+  //     link.click();
+  //     link.remove();
+  //   };
+  // };
```
Comment (afc163) [inline components/image/demo/toolbarRender.tsx:68]: 不需要注释，可以放一个 codesandbox 单独展示这个能力。

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx:68]: 要等 antd minor 发版才行吧？不然现在 codesandbox 里的 antd 版本没有这些 api

Comment (linxianxi) [reply components/image/demo/toolbarRender.tsx]: 已经先删了，发版了再加上来

**Review by zombieJ — APPROVED** [review]

**Pull request merged**